### PR TITLE
task: combine APR into one item

### PR DESCRIPTION
**Pull request by amytsang**
Screenshot 2024-04-23 at 8 49 32 AM
Screenshot 2024-04-23 at 7 07 11 AM

- Remove Lend APY and Rewards APR column. Combine them into Total APR column, add Tooltip to show breakdown.
- Reduce Supply market list size.

Screenshot 2024-04-23 at 7 07 55 AM

- Update Rewards tAPR section to Total APR

Screenshot 2024-04-23 at 7 12 05 AM

- Update text Fee to AMM swap fee

Screenshot 2024-04-23 at 7 13 27 AM
Screenshot 2024-04-23 at 7 13 33 AM

-Update mobile view with Total APR

Screenshot 2024-04-23 at 7 14 25 AM

- Update Lend APY to Lend APR

--- 
- Fixed User supply list filter not showing properly.
- Add search params routes for market list when navigating between networks.

### CODE REVIEW

File: apps/lend/src/hooks/useSupplyTotalApr.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  const { rates } = marketRatesResp ?? {}
+
+  const lendApr = +(rates?.lendApr || '0')
+  const lendApy = +(rates?.lendApy || '0')
+  const [crvBase = 0, crvBoost = 0] = crv ?? []
+  const others = other ?? []
+
+  resp.totalApr = _getTotalApr(lendApr, crvBase, crvBoost, others)
+  resp.tooltipValues = _getTooltipValue(lendApr, lendApy, crvBase, crvBoost, others)
+  return resp
```
Comment (DanielSchiavini) [inline apps/lend/src/hooks/useSupplyTotalApr.ts:88]:
```suggestion
  return {
    totalApr: _getTotalApr(lendApr, crvBase, crvBoost, others),
    tooltipValues: _getTooltipValue(lendApr, lendApy, crvBase, crvBoost, others),
  }
```

File: apps/lend/src/hooks/useSupplyTotalApr.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      crvBoosted: string
+      incentives: string[]
+      incentivesObj: RewardOther[]
+    } | null
+  } = {
+    totalApr: { min: '', max: '', minMax: '' },
+    tooltipValues: null,
+  }
+
+  if (typeof marketRewardsResp === 'undefined' || typeof marketRatesResp === 'undefined') return resp
```
Comment (DanielSchiavini) [inline apps/lend/src/hooks/useSupplyTotalApr.ts:76]: The default value only gets used here, we could inline it for clarity
By returning the data directly we don't need to declare the types
```suggestion
  if (!marketRewardsResp || !marketRatesResp) {
    return {
      totalApr: { min: '', max: '', minMax: '' },
      tooltipValues: null,
    }
  }
```

File: apps/lend/src/layout/Header.tsx [hunk 3]
```diff
@@ -117,7 +117,15 @@ const Header = () => {
   const handleNetworkChange = (selectedChainId: React.Key) => {
     if (rChainId !== selectedChainId) {
       const network = networks[selectedChainId as ChainId].id
-      navigate(`${rLocalePathname}/${network}/${getRestPartialPathname()}`)
+      const [currPath] = window.location.hash.split('?')
+
+      if (currPath.endsWith('markets')) {
```
Comment (DanielSchiavini) [inline apps/lend/src/layout/Header.tsx:122]: It is generally useful to keep page URLs in constants so they can be updated without introducing bugs

File: apps/lend/src/lib/apiLending.ts [hunk 4]
```diff
@@ -319,61 +319,74 @@ const market = {
...
+          other: [],
+          crv: [0, 0],
+        },
+        error: '',
+      }
+
+      // check if gauge is valid
+      if (owm.addresses.gauge === INVALID_ADDRESS) {
+        return resp
+      } else {
```
Comment (DanielSchiavini) [inline apps/lend/src/lib/apiLending.ts:334]: else is not necessary

**Review by Macket — APPROVED** [review]

**Review by bout3fiddy — APPROVED** [review]

**Review by michwill — APPROVED** [review]

**Pull request merged**